client/signup/steps/plans/util.ts

```diff
@@ -1,15 +1,19 @@
 import { getUrlParts } from '@automattic/calypso-url';
+import type { IntervalType } from 'calypso/my-sites/plans-features-main/types';
 
 type DomainItem = {
 	is_domain_registration?: boolean;
 	meta?: string;
 	product_slug?: string;
 };
 
-export const getIntervalType = ( path?: string ): string => {
+// TODO: this can be so much simpler once all the call sites of this functions are TS
```

Comment:
Good question. My original thought was actually wrong: I was thinking if we could restrict the type of `path` as `IntervalType` to limit the potential value somehow from the call sites, this function could be simpler. However, that doesn't make sense since type conversion would just happen elsewhere. I've removed this misleading comment.